src/components/tl/mlx5/tl_mlx5.c

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2022.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_mlx5.h"
+
+ucc_status_t ucc_tl_mlx5_get_lib_attr(const ucc_base_lib_t *lib,
+                                      ucc_base_lib_attr_t * base_attr);
+ucc_status_t ucc_tl_mlx5_get_context_attr(const ucc_base_context_t *context,
+                                          ucc_base_ctx_attr_t *     base_attr);
+
+static ucc_config_field_t ucc_tl_mlx5_lib_config_table[] = {
+    {"", "", NULL, ucc_offsetof(ucc_tl_mlx5_lib_config_t, super),
+     UCC_CONFIG_TYPE_TABLE(ucc_tl_lib_config_table)},
+
+    {"ASR_BARRIER", "0", "Boolean - use  service barrier or p2p sync of ASRs",
+     ucc_offsetof(ucc_tl_mlx5_lib_config_t, asr_barrier), UCC_CONFIG_TYPE_UINT},
+
+    {"DM_BUF_SIZE", "8k", "Size of the pre-allocated DeviceMemory buffer",
+     ucc_offsetof(ucc_tl_mlx5_lib_config_t, dm_buf_size),
+     UCC_CONFIG_TYPE_MEMUNITS},
+
+    {"DM_BUF_NUM", "auto", "Number of DM buffers to alloc",
+     ucc_offsetof(ucc_tl_mlx5_lib_config_t, dm_buf_num),
+     UCC_CONFIG_TYPE_MEMUNITS},
```

Comment:
is UCC_CONFIG_TYPE_MEMUNITS is correct here?

Reply:
no, should be UINT
fixed